index.bs

```diff
@@ -4894,16 +4896,16 @@ function process(numberOfFrames) {
 
 	// Render each sample frame in the quantum
 	for (let index = 0; index &lt; numberOfFrames; index++) {
-		// Check that currentTime is within allowable range for playback
-		if (currentTime &lt; start || currentTime >= stop) {
+		// Check that currentTime and bufferDuration are within allowable range for playback
+		if (currentTime &lt; start || currentTime >= stop || bufferDuration >= duration) {
 			output.push(0); // this sample frame is silent
 			currentTime += dt;
 			continue;
 		}
 
 		if (!started) {
 			// Take note that buffer has started playing and get initial playhead position.
-			bufferTime = offset + ((currentTime - start) * computedPlaybackRate);
+			bufferTime = offset + bufferDuration;
```

Comment:
Is this right?  The initial value for bufferTime is now different between this version and the old version. I think the original is the right value.

Reply:
You're correct about the inconsistency. But another way to go would be to initialize `bufferDuration` to `(currentTime - start) * computedPlaybackRate` in this clause. Wouldn't that be an even better fix? It would cause `bufferDuration` to respect fractional-frame start times.

Reply:
I noticed this change too, but thought it was fixing a bug.

For the `when` parameter, "If 0 is passed in for this value or if the value is
less than currentTime, then the sound will start playing immediately."  Also,
"The `offset` parameter supplies a playhead position where playback will begin."
I expect playback to begin with the head at `offset` regardless of `when`.

I couldn't make sense of ((currentTime - start) * computedPlaybackRate).
It seems to be aiming to make playback start with the head at some point
after `offset` if `when` is less than `currentTime`.

(The spec text is helpful to indicate the intention of the algorithm.)

Reply:
Yeah, I didn't look to see what the original is doing, but I agree with @karlt: playback should start at `offset` independent of `when`.  I think `bufferDuration` should actually start at 0.  If you choose any other value, then the actual playback duration will be less then the `duration` arg, even if playback rate is 1.

Reply:
`bufferDuration` starts at 0.  Simply initializing `bufferTime` to `offset` would be more clear.

Reply:
@rtoy could you please change this as per @karlt's last suggestion? I will not be able to work on this for a couple of weeks and this PR is close to done.